docs/api.md

```diff
@@ -0,0 +1,287 @@
+
+```
+{"token-pair":
+   {"auth-token":"eyJh..........VdbpA",
+    "refresh-token":"eyJhb........sr4A"
+   }
+}
+```
+
+## How the receipt mechanism works.
+
+The Witan API is based on an asynchronus system. For aspects like uploads for example the API will issue you a receipt id. You are expected to use this receipt along with your auth token to query the API until the action is complete.
+
+For example: you want to upload a file, once you've authenticated your user and have an auth token you will then call `/api/files/upload` with your auth token as a header. In response the API will send back a `receipt-id` value and a 202 response code. 
+
+While Witan processes your file upload request you are expected to use the endpoint `/api/receipts/[receipt-id]` to see if your upload request url is ready. If Witan has successfully processed created an upload link you will receive a 200 response code along with upload link and filestore id. 
+
+Once you have a 200 response and the required url/id combination you can safely upload your file. 
+
+## How to upload a file.
+
+The file upload aspect of the Witan API is not uploading files in the traditional sense. The API provides a mechanism to generate a URL and file id to upload a file to. The actual uploading of the file, and how that is executed, is up to you. 
+
+There are three main steps to uploading a file and its metadata to Witan via the API.
+
+### Step 1 - Generate your receipt-id.
+
+Assuming that you have logged in and have an auth token the first step is to do a POST to the `/api/file/upload` endpoint. 
+
+```
+curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header 'authorization: eyJhb.....7SeMQ' 'https://api.witanforcities.com/api/files/upload'
+```
+
+This will return a receipt-id, using this id you can now poll against the API to get the file upload link.
+
+```
+curl -X GET --header 'Accept: application/json' --header 'authorization: eyJhb.....7SeMQ' 'https://api.witanforcities.com/api/receipts/abcdefg-abcd-abcd-abcd-abcdeghij'
+```
+
+### Step 2 - Get the upload link.
+
+The response from the `/api/receipt/[receipt-id]` call will respond with either a 202 code, meaning the process is still being handled by Witan, you should continue to call the receipt endpoint with the same URL to see if the receipt has updated. 
+
+If you receive a response with a 200 status code then the request is classed as complete and you will see the following response from the API. 
+
+```
+{
+  "witan.httpapi.spec/uri": "https://prod-witan-kixi-datastore-filestore...........",
+  "kixi.datastore.filestore/id": "d1bec41e.....46c2"
+}
+```
+
+It is your responsibility to upload the file you want stored on Witan using the URL that is in the `witan.httpapi.spec/url` response field. 
+
+```
+curl -vvv -XPUT --data-binary /Users/jasonbell/Downloads/resp_gas.xml "https://prod-witan-kixi-datastore-filestore.s3-eu-west-1.amazonaws.com/d1bec41e-086e-4ad7-
```

Comment:
Is there a way to do this part on Swagger?

Reply:
line 108 should be `.spec/uri` rather than `.spec/url`

Reply:
Might be good to explicitly drawn attention to the fact that the upload data needs to be binary. This prohibits the use of the standard `requests.post(url, data=file)` method in python and it through me for a bit of a loop. A reminder about the binary file outside of the curl code might have made this a bit easier to solve.

Reply:
Good catch Mike on the uri/url error. I've added a line on uploading as binary data. Seb: I'm not aware that you'd be able to do this on Swagger. The Swagger tool is really there for reference and trying out, not doing actual work with.